orchestrator/schema/property.py

```diff
@@ -84,30 +177,56 @@ class AbstractProperty(Property):
     concretePropertyIdentifiers: list[str] | None = None
     model_config = ConfigDict(frozen=True)
 
+    @classmethod
+    def from_descriptor(cls, descriptor: AbstractPropertyDescriptor):
+
+        return cls(
+            identifier=descriptor.identifier,
+        )
+
     def __str__(self):
         return f"ap-{self.identifier}"
 
     def __eq__(self, other):
 
         retval = super().__eq__(other)
         if retval:
-            retval = (
-                self.concretePropertyIdentifiers == other.concretePropertyIdentifiers
-            )
+            try:
+                retval = (
+                    self.concretePropertyIdentifiers
+                    == other.concretePropertyIdentifiers
+                )
+            except AttributeError:
+                retval = False
 
         return retval
```

Comment:
This could be something like

```python
return super().__eq__(other) and hasattr(other, "concretePropertyIdentifiers") and self.concretePropertyIdentifiers == other.concretePropertyIdentifiers
```

(this likely needs to be formatted and checked)